nav2_bt_navigator/src/bt_navigator.cpp

```diff
@@ -241,109 +140,65 @@ nav2_util::CallbackReturn
 BtNavigator::on_shutdown(const rclcpp_lifecycle::State & /*state*/)
 {
   RCLCPP_INFO(get_logger(), "Shutting down");
+
+  if (!bt_action_server_->on_shutdown()) {
+    return nav2_util::CallbackReturn::FAILURE;
+  }
+
   return nav2_util::CallbackReturn::SUCCESS;
 }
 
-void
-BtNavigator::navigateToPose()
+bool
```

Comment:
The `navigateToPose` -> `on_goal_received` and `initializeGoalPose` -> `on_loop` changes are honestly a little hard for me to fully register. There's a bunch of stuff reordered that makes it hard for me to try to draw parallels. Can you highlight if these functions roles change at all or if its just doing less stuff? Any reason why some of it seems like its been reordered within the same function?

Reply:
`on_goal_received` basically combines some of the things that were happening in `navigateToPose` and `initializeGoalPose`. Earlier, `navigateToPose` was the entry point when a new goal was received and `initializeGoalPose` was the first thing called in `navigateToPose` to reset some state variables and put the new navigation goal on the blackboard. `on_goal_received` replaces this logic since it's now the entry point and the place where state variables should be reset.

After that, `navigateToPose` defined the lamdas `on_loop` and `is_cancelling` and waited for the BT to finish ticking. This logic has been moved to the abstract server.

`on_loop` basically takes out some BT Navigator-specific logic that was earlier defined in the lambda within `navigateToPose`. It's not directly related to either `navigateToPose` or `initializeGoalPose`. This is where a user can define the execution that happens on one iteration through the BT (things like publishing feedback, etc.).